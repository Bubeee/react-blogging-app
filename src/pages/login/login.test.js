import React from 'react';
import { Login } from '.';

describe('Login', () => {
  it('should render correctly', () => {
    const output = shallow(<Login />);

    expect(output).toMatchSnapshot();
  });

  it('should call submit on submit', () => {
    const mockedEvent = { target: {}, preventDefault: () => null }

    const wrapper = shallow(<Login />);
    wrapper.instance().submit = jest.fn();
    wrapper.update();

    wrapper.find('form').simulate('submit', mockedEvent);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.instance().submit).toBeCalledWith(mockedEvent);
  });
});
