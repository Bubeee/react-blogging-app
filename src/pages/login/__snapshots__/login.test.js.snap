// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should call submit on submit 1`] = `
<div>
  <Header />
  <div
    className="login-form__container"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="post-form__label"
        htmlFor="title"
      >
        Email
      </label>
      <input
        className="login-form__input"
        name="title"
        placeholder="Enter your email"
        type="text"
      />
      <label
        className="post-form__label"
        htmlFor="content"
      >
        Password
      </label>
      <input
        className="login-form__password"
        name="content"
        placeholder="Enter your password"
        type="password"
      />
      <input
        className="login-form__submit"
        type="submit"
        value="Login"
      />
    </form>
  </div>
  <Footer />
</div>
`;

exports[`Login should render correctly 1`] = `
<div>
  <Header />
  <div
    className="login-form__container"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="post-form__label"
        htmlFor="title"
      >
        Email
      </label>
      <input
        className="login-form__input"
        name="title"
        placeholder="Enter your email"
        type="text"
      />
      <label
        className="post-form__label"
        htmlFor="content"
      >
        Password
      </label>
      <input
        className="login-form__password"
        name="content"
        placeholder="Enter your password"
        type="password"
      />
      <input
        className="login-form__submit"
        type="submit"
        value="Login"
      />
    </form>
  </div>
  <Footer />
</div>
`;
